extensions/security/deployment/src/main/java/io/quarkus/security/deployment/SecurityProcessor.java

```diff
@@ -590,8 +593,12 @@ void registerAdditionalBeans(BuildProducer<AdditionalBeanBuildItem> beans) {
     }
 
     @BuildStep
-    AdditionalBeanBuildItem authorizationController() {
-        return AdditionalBeanBuildItem.builder().addBeanClass(AuthorizationController.class).build();
+    AdditionalBeanBuildItem authorizationController(LaunchModeBuildItem launchMode) {
+        Class<? extends AuthorizationController> controllerClass = AuthorizationController.class;
+        if (launchMode.getLaunchMode() == LaunchMode.DEVELOPMENT && !security.authorizationEnabled) {
+            controllerClass = DisabledAuthorizationController.class;
+        }
+        return AdditionalBeanBuildItem.builder().addBeanClass(controllerClass).build();
```

Comment:
An alternative would be to define the property as a property which can be accessed at runtime and then use that in the controller.

Reply:
I feel we should not provide a runtime level support for disabling the authorization at runtime